lib/compilers/WSL-CL.js

```diff
@@ -64,4 +66,17 @@ function compileCl(info, env) {
     return compile.initialise();
 }
 
+// AP: Create CE temp directory in winTmp directory. This code matches overrides the function in base-compiler.js. 
+// NPM temp package: https://www.npmjs.com/package/temp, see Affixes
+Compile.prototype.newTempDir = function () {
```

Comment:
So the override mechanism works like this:
The compiler specifies a type on the properties, which is used to load a `compiler/TYPE.js` file, and we get whatever it outputs as a kind of "wrapper" around the compiler.
Now, insde the function that gets exported, you initialize a new Compile object, in which you change the objects that need to be changed, but only to that Compile object, not the global one (As you're doing right now)

So you need to move this `Compiler.prototype.newTempDir` function inside the compileCl function (after L37 `var compile ...` but before the initialization at L66) and overrride the newTempDir like `compile.newTempDir = function () { ... };`
I hope some of this made any sense

Reply:
Yes, that makes sense. I still need to read the code more carefully, but I've pushed the changes. And my printf debugging says you're correct as well :)